src/server/replica.cc

```diff
@@ -219,8 +218,8 @@ void Replica::MainReplicationFb() {
     if ((state_mask_ & R_GREETED) == 0) {
       ec = Greet();
       if (ec) {
-        LOG(INFO) << "Error greeting " << ec;
-        state_mask_ &= ~R_TCP_CONNECTED;
+        LOG(INFO) << "Error greeting " << ec << " " << ec.message();
+        state_mask_ &= R_ENABLED;
```

Comment:
this change has the same affect as ~R_TCP_CONNECTED because the only flag enabled in this state is R_TCP_CONNECTED and R_ENABLED, but this change makes it more readable and if we add new state between connect and greet we will not need to do any change here so this is more robust